src/auth/mas.py

```diff
@@ -2,20 +2,155 @@
 # See LICENSE file for licensing details.
 
 """Helper module used to manage MAS-related workloads."""
-
 import logging
+import secrets
 
+import ops
 from jinja2 import Environment, FileSystemLoader, select_autoescape
 
 from state.charm_state import SynapseConfig
 from state.mas import MASConfiguration
 
+logger = logging.getLogger()
+
 MAS_TEMPLATE_FILE_NAME = "mas_config.yaml.j2"
 
-logger = logging.getLogger()
+MAS_SERVICE_NAME = "synapse-mas"
+MAS_EXECUTABLE_PATH = "/usr/bin/mas-cli"
+MAS_WORKING_DIR = "/mas"
+MAS_CONFIGURATION_PATH = f"{MAS_WORKING_DIR}/config.yaml"
+
+MAS_PEBBLE_LAYER = ops.pebble.LayerDict(
+    {
+        "summary": "Matrix Authentication Service layer",
+        "description": "pebble config layer for MAS",
+        "services": {
+            MAS_SERVICE_NAME: {
+                "override": "replace",
+                "summary": "Matrix Authentication Service",
+                "startup": "enabled",
+                "command": f"{MAS_EXECUTABLE_PATH} server -c {MAS_CONFIGURATION_PATH}",
+                "working-dir": MAS_WORKING_DIR,
+            }
+        },
+    }
+)
+
+
+class MASConfigInvalidError(Exception):
```

Comment:
I wonder if we really need different exceptions for ops.pebble.ExecError. Are they handled differently?